src/daemon/rpmostreed-daemon.cxx

```diff
@@ -737,10 +737,21 @@ char *
 rpmostreed_daemon_client_get_sd_unit (RpmostreedDaemon *self, const char *client)
 {
   auto clientdata = static_cast<struct RpmOstreeClient *>(g_hash_table_lookup (self->bus_clients, client));
-  if (!clientdata || clientdata->sd_unit == NULL)
-    return NULL;
-  else
-    return g_strdup (clientdata->sd_unit);
+  /* If the caller didn't register (e.g. Zincati doesn't today because it runs as a
+   * separate UID which systemd doesn't consider `active`), then let's look up its
+   * systemd unit now.
+   */
+  if (!clientdata)
+    {
+      pid_t pid;
+      char *sd_unit = NULL;
+      if (!get_client_pid (self, client, &pid))
+        return NULL;
+      if (sd_pid_get_user_unit (pid, &sd_unit) != 0)
```

Comment:
Minor: Looking at the actual implementation, I think this is correct, though the manpage says:

> On success, these calls return 0 or a positive integer. On failure, these calls return a negative errno-style error code.

So it would technically be more appropriate to do `< 0` instead.

Reply:
thanks!